selftests/functional/test_basic.py

```diff
@@ -476,6 +477,21 @@ def test_invalid_python(self):
         self.assertIn('1-%s:MyTest.test_my_name -> TestError' % test,
                       result.stdout)
 
+    @unittest.skipIf(process.system("which /bin/read", ignore_status=True),
+                     "/bin/read not available.")
+    def test_read(self):
+        try:
+            cmd = utils_path.find_command("reada")
```

Comment:
Typo: `s/reada/read/`.

Also, I don't get:
1) why you're trying to skip the test from inside the test itself
2) there's also a unittest `skipIf` decorator